include/deal.II/lac/generic_linear_algebra.h

```diff
@@ -68,10 +68,51 @@ namespace LinearAlgebraPETSc
 {
   using namespace dealii;
 
+  /**
+   * Typedef for the base class for preconditioners.
+   */
+  typedef PETScWrappers::PreconditionerBase PreconditionerBase;
+
+  /**
+   * Typedef for the base class for linear solvers.
+   */
+  typedef PETScWrappers::SolverBase SolverBase;
+
+  /**
+   * Typedef for the BiCGStab linear solver.
+   */
+  typedef PETScWrappers::SolverBicgstab SolverBicgstab;
+
+  /**
+   * Typedef for the CG linear solver.
+   */
   typedef PETScWrappers::SolverCG SolverCG;
+
+  /**
+   * Typedef for the CG Squared linear solver.
+   */
+  typedef PETScWrappers::SolverCGS SolverCGS;
+
+#ifdef DEAL_II_PETSC_WITH_MUMPS
+
+  /**
+   * Typedef for the direct linear solver (via MUMPS).
+   */
+  typedef PETScWrappers::SparseDirectMUMPS SolverDirect;
```

Comment:
Fair enough. It was the only option for a direct solver for `PETSc`, so I though it made sense. I'll do as you've suggested and, if a generic direct solver interface is created for `PETSc`, we can then add it to the list.